File: Homework1/fourargs.cpp
#include <iostream>

int i;
// Copyright year Simin Zhai siminz@bu.edu

int ot=0;

int main(int argumentcount,  char **arguments) {
	if (argumentcount > 5)
	{
		ot = 5;
	}	
	else
	{
		ot = argumentcount;
	}

	for (i=1; i < ot; i++)
	{
		std::cout << arguments[i] <<"\n";
	}
	for (i=5; i < argumentcount; i++)
	{	
		std::cerr << arguments[i] <<"\n";
	}
	
	return 0;
}

File: Homework1/fourargs.py
# Copyright year Simin Zhai siminz@bu.edu

import sys 

#sys.stdout = open('stdout', 'w')
#sys.stderr = open('stderr', 'w')

ot = 0

if len(sys.argv) > 5: 
	ot = 5
else:
	ot = len(sys.argv)

for i in range(1,ot):
	sys.stdout.write(sys.argv[i]+ '\n')


for i in range(5,len(sys.argv)):
	sys.stderr.write(sys.argv[i]+ '\n')
